src/mono/cmake/configure.cmake

```
@@ -85,21 +85,28 @@ ac_check_funcs (
   gethrtime read_real_time gethostbyname gethostbyname2 getnameinfo getifaddrs
   access inet_ntop Qp2getifaddrs getpid mktemp)
 
-if (HOST_LINUX OR HOST_BROWSER)
+if (HOST_LINUX OR HOST_BROWSER OR HOST_WASI)
   # sysctl is deprecated on Linux and doesn't work on Browser
   set(HAVE_SYS_SYSCTL_H 0)
 else ()
   check_include_files("sys/types.h;sys/sysctl.h" HAVE_SYS_SYSCTL_H)
 endif()
 
+if (HOST_WASI)
+  # sysctl is deprecated on Linux and doesn't work on Browser
+  set(HAVE_GETRUSAGE 0)
+endif()
+
 check_include_files("sys/types.h;sys/user.h" HAVE_SYS_USER_H)
 
 if(NOT HOST_DARWIN)
   # getentropy was introduced in macOS 10.12 / iOS 10.0
   ac_check_funcs (getentropy)
 endif()
 
-find_package(Threads)
+if(NOT HOST_WASI)
```

not sure if this works but might be clearer:
```suggestion
if(NOT DISABLE_THREADS)
```